### DISCUSSION TITLE
only Vsync, Update rate is capped at 4x refresh late

### DISCUSSION THREAD

**Discussion by Hinaxnitro** [ppy/osu#30886] [Q&A] [answered]
When I set the frame limit to Vsync, the Update rate is capped at 240fps on a 60Hz monitor.

When I assume Vsync's Draw limit to display's refresh late, it's weird since other settings limit it to 2 times the Draw limit.

Is there any reason for this?

**Comment by peppy** [comment]
Having a higher update rate results in lower latency when we are running multi-threaded. This is due to the alignment of the finishing times of update and draw threads. Running the update thread at a higher rate will reduce the delay by 50% on average.

If you want to avoid this for performance / heat reasons, you can change the game thread mode to "single threaded" from settings and you'll get what you expect to see.

**Reply by Hinaxnitro** [reply]
I understand that the app needs the Update rate higher than the Draw rate.

What I wanted to ask is why Vsync's Update rate is 4 times of Draw rate while other settings are 2 times of Draw rate.
Your explanation feels that apps need even more Update rate since the monitor's timing is not stable enough.

Am I getting right?

**Reply by peppy** [reply, marked as answer]
it's probably set this way because running at vsync generally means you're at a much lower frame rate, so having higher update rate is helpful.
